source/version.js

```diff
@@ -1,27 +1,50 @@
 'use strict';
 const semver = require('semver');
 
-exports.SEMVER_INCREMENTS = ['patch', 'minor', 'major', 'prepatch', 'preminor', 'premajor', 'prerelease'];
-exports.PRERELEASE_VERSIONS = ['prepatch', 'preminor', 'premajor', 'prerelease'];
+class Version {
+	constructor(version) {
+		this.version = version;
+	}
 
-const isValidVersion = input => Boolean(semver.valid(input));
+	isPrereleaseVersion() {
+		return module.exports.PRERELEASE_VERSIONS.includes(this.version) || Boolean(semver.prerelease(this.version));
+	}
 
-exports.isValidVersionInput = input => exports.SEMVER_INCREMENTS.includes(input) || isValidVersion(input);
+	getNewVersion(input) {
+		module.exports.validateVersion(this.version);
+		if (!module.exports.isValidVersionInput(input)) {
+			throw new Error(`Version should be either ${module.exports.SEMVER_INCREMENTS.join(', ')} or a valid semver version.`);
+		}
 
-exports.isPrereleaseVersion = version => exports.PRERELEASE_VERSIONS.includes(version) || Boolean(semver.prerelease(version));
+		return module.exports.SEMVER_INCREMENTS.includes(input) ? semver.inc(this.version, input) : input;
+	}
 
-exports.getNewVersion = (oldVersion, input) => {
-	if (!exports.isValidVersionInput(input)) {
-		throw new Error(`Version should be either ${exports.SEMVER_INCREMENTS.join(', ')} or a valid semver version.`);
+	isGreaterThanOrEqualTo(otherVersion) {
+		module.exports.validateVersion(this.version);
+		module.exports.validateVersion(otherVersion);
+
+		return semver.gte(otherVersion, this.version);
 	}
 
-	return exports.SEMVER_INCREMENTS.includes(input) ? semver.inc(oldVersion, input) : input;
-};
+	isLowerThanOrEqualTo(otherVersion) {
+		module.exports.validateVersion(this.version);
+		module.exports.validateVersion(otherVersion);
 
-exports.isVersionGreater = (oldVersion, newVersion) => {
-	if (!isValidVersion(newVersion)) {
-		throw new Error('Version should be a valid semver version.');
+		return semver.lte(otherVersion, this.version);
 	}
+}
 
-	return semver.gt(newVersion, oldVersion);
+module.exports = version => new Version(version);
+
+module.exports.SEMVER_INCREMENTS = ['patch', 'minor', 'major', 'prepatch', 'preminor', 'premajor', 'prerelease'];
+module.exports.PRERELEASE_VERSIONS = ['prepatch', 'preminor', 'premajor', 'prerelease'];
+
+const isValidVersion = input => Boolean(semver.valid(input));
+
+module.exports.isValidVersionInput = input => module.exports.SEMVER_INCREMENTS.includes(input) || isValidVersion(input);
```

Comment:
Should this method be renamed to `isValidInput`?

Reply:
Yes